Improve checks for latex support

andreas-bok-sociomantic opened the pull request:
It is quite frustrating build the `pdf` manual for **libdrizzle-redux** if not all the required `latex` packages are installed. 

This PR has the following parts 
- adds `autoconf` macros which checks for the programs, `latex`, `pdflatex` and `kpsewhich` 
- adds a **bash** script which tests whether a required package is available using `kpsewhich`

ben-palmer-sociomantic commented on m4/acltx_prog_kpsewhich.m4:
```diff
@@ -0,0 +1,20 @@
+dnl This macro find a kpsewhich application and set the variable
+dnl kpsewhich to the name of the application or to no if not found if
+dnl ACTION-IF-NOT-FOUND is not specified, configure fail when then
+dnl application is not found.
```
This whole paragraph is very strangely worded. Perhaps it should be something more like:

```
This macro looks for the kpsewhich application and sets the variable
kpsewhich to the name of the application if found or no if not found. If
ACTION-IF-NOT-FOUND is not specified when application is not found then fail.
```

ben-palmer-sociomantic replied:
I still thing this paragraph is strangely worded.

ben-palmer-sociomantic commented on m4/acltx_prog_latex.m4:
```diff
@@ -0,0 +1,52 @@
+#   This macro find a latex application and set the variable latex to
+#   the name of the application or to no if not found if
+#   ACTION-IF-NOT-FOUND is not specified, configure fail when then
+#   application is not found.
```
Again this paragraph is strangely worded.

ben-palmer-sociomantic replied:
Same here

ben-palmer-sociomantic commented on m4/acltx_prog_latex.m4:
```diff
@@ -0,0 +1,52 @@
+#   ACTION-IF-NOT-FOUND is not specified, configure fail when then
+#   application is not found.
+#
+#   It is possible to set manually the program to use using latex=...
```
should be `is possible to manually set`

ben-palmer-sociomantic commented on m4/acltx_prog_pdflatex.m4:
```diff
@@ -0,0 +1,20 @@
+dnl This macro find a pdflatex application and set the variable
+dnl pdflatex to the name of the application or to no if not found if
+dnl ACTION-IF-NOT-FOUND is not specified, configure fail when then
+dnl application is not found.
```
Again this paragraph needs rewording.

ben-palmer-sociomantic replied:
Same here.

ben-palmer-sociomantic commented:
The wording of the commit title and message for the commit `Use variable for description` are a bit unclear. The title could be more clear so maybe `Use variable for documentation description` or something like it would be more specific. The message is not grammatically correct, perhaps it should be `Store description as a variable instead of as a string literal`.

andreas-bok-sociomantic commented:
> Again this paragraph is strangely worded.

My bad, didn't actually read through the autoconf macro files, since they are added directly from the autoconf repo, http://ac-archive.sourceforge.net/ac-archive/

I will change review them, and also add a PR against the autoconf repo.

andreas-bok-sociomantic commented:
Reworded Yoda speech 

```diff
diff --git a/m4/acltx_prog_kpsewhich.m4 b/m4/acltx_prog_kpsewhich.m4
index c73b4b6..fc68295 100644
--- a/m4/acltx_prog_kpsewhich.m4
+++ b/m4/acltx_prog_kpsewhich.m4
@@ -1,9 +1,10 @@
 dnl @synopsis ACLTX_PROG_KPSEWHICH([ACTION-IF-NOT-FOUND])
 dnl
-dnl This macro find a kpsewhich application and set the variable
-dnl kpsewhich to the name of the application or to no if not found if
-dnl ACTION-IF-NOT-FOUND is not specified, configure fail when then
-dnl application is not found.
+dnl Tests if the application kpsewhich exists. If so, the variable $kpsewhich is
+dnl assigned the name of the application, or 'no' if kpsewhich was not found.
+dnl    In the latter case, configure will fail, if ACTION-IF-NOT-FOUND is not
+dnl    specified
+dnl
 dnl
 dnl @category LaTeX
 dnl @author Boretti Mathieu <boretti@eig.unige.ch>
diff --git a/m4/acltx_prog_latex.m4 b/m4/acltx_prog_latex.m4
index 9563740..8e998e7 100644
--- a/m4/acltx_prog_latex.m4
+++ b/m4/acltx_prog_latex.m4
@@ -6,12 +6,12 @@
 #
 # DESCRIPTION
 #
-#   This macro find a latex application and set the variable latex to
-#   the name of the application or to no if not found if
-#   ACTION-IF-NOT-FOUND is not specified, configure fail when then
-#   application is not found.
+#   Check if a latex application exists and set the variable $latex
+#   to the name of the application or to 'no' if none was found.
+#      In the latter case, configure will fail, if ACTION-IF-NOT-FOUND is not
+#      specified
 #
-#   It is possible to set manually the program to use using latex=...
+#   It is possible to manually set the program using latex=...
 #
 # LAST MODIFICATION
 #
diff --git a/m4/acltx_prog_pdflatex.m4 b/m4/acltx_prog_pdflatex.m4
index 7781e59..6d543c8 100644
--- a/m4/acltx_prog_pdflatex.m4
+++ b/m4/acltx_prog_pdflatex.m4
@@ -1,9 +1,9 @@
-dnl @synopsis AC_PROG_PDFLATEX
+dnl @synopsis ACLTX_PROG_PDFLATEX([ACTION-IF-NOT-FOUND])
 dnl
-dnl This macro find a pdflatex application and set the variable
-dnl pdflatex to the name of the application or to no if not found if
-dnl ACTION-IF-NOT-FOUND is not specified, configure fail when then
-dnl application is not found.
+dnl Tests if the pdflatex application exists and sets the variable $pdflatex to
+dnl the name of the application or to 'no', if it was not found.
+dnl    In the latter case, configure will fail, if ACTION-IF-NOT-FOUND is not
+dnl    specified
 dnl
 dnl @category LaTeX
 dnl @author Boretti Mathieu <boretti@eig.unige.ch>

```

ben-palmer-sociomantic commented on docs/conf.py.in:
```diff
@@ -188,7 +188,8 @@ htmlhelp_basename = 'gearmandoc'
 latex_documents = [
-  ('index', 'libdrizzle.tex', description, u'Drizzle Development Group', 'manual'),
+  ('index', 'libdrizzle.tex', description,
+   u'Drizzle Development Group', 'manual'),
```
I don't really get what this change is doing based on the commit message.

ben-palmer-sociomantic commented on docs/latex_check_packages.sh:
```diff
@@ -0,0 +1,97 @@
+##
+# shell script which checks whether required packages
+# is available for building the pdf documentation
```
Should be `required packages are available`.

ben-palmer-sociomantic commented on m4/acltx_prog_kpsewhich.m4:
```diff
@@ -1,9 +1,10 @@
-dnl kpsewhich to the name of the application or to no if not found if
-dnl ACTION-IF-NOT-FOUND is not specified, configure fail when then
-dnl application is not found.
+dnl Tests if the application kpsewhich exists. If so, the variable $kpsewhich is
```
Ah I didn't see this commit when I made my other comments. You can ignore them.

ben-palmer-sociomantic reviewed (commented):
I made a couple of comments bu hadn't seen the last commit so some of them can be ignored.

andreas-bok-sociomantic committed:
Check for latex support

Modify configure.ac to include checks for
the programs: latex, pdflatex and kpsewhich

Only emits warnings if the check fails, since
the latex support is not a hard requirement

andreas-bok-sociomantic committed:
Add check for required latex packages

Adds bash script which checks for latex packages
used by pdflatex to generate the pdf manual

The check is used with the new make target:
    latex-package-check

andreas-bok-sociomantic commented:
Addressed comments and squashed commits
```
 # The name of an image file (relative to this directory) to place at the top of
diff --git a/docs/latex_check_packages.sh b/docs/latex_check_packages.sh
index a4d62b7..a6691d3 100755
--- a/docs/latex_check_packages.sh
+++ b/docs/latex_check_packages.sh
@@ -1,6 +1,6 @@
 ##
 # shell script which checks whether required packages
-# is available for building the pdf documentation
+# are available for building the pdf documentation
 #
```

andreas-bok-sociomantic commented:
The changes on `docs/conf.py.in` reverted an earlier commit removing the file completely from this PR

ben-palmer-sociomantic commented on m4/acltx_prog_kpsewhich.m4:
```diff
@@ -0,0 +1,21 @@
+dnl Tests if the application kpsewhich exists. If so, the variable $kpsewhich is
+dnl assigned the name of the application, or 'no' if kpsewhich was not found.
+dnl	In the latter case, configure will fail, if ACTION-IF-NOT-FOUND is not
+dnl	specified
```
Is there some weird indentation here?

ben-palmer-sociomantic commented on m4/acltx_prog_latex.m4:
```diff
@@ -0,0 +1,52 @@
+#   Check if a latex application exists and set the variable $latex
+#   to the name of the application or to 'no' if none was found.
+#	In the latter case, configure will fail, if ACTION-IF-NOT-FOUND is not
+#	specified
```
Is there some weird indentation here?

ben-palmer-sociomantic commented on m4/acltx_prog_pdflatex.m4:
```diff
@@ -0,0 +1,20 @@
+dnl Tests if the pdflatex application exists and sets the variable $pdflatex to
+dnl the name of the application or to 'no', if it was not found.
+dnl	In the latter case, configure will fail, if ACTION-IF-NOT-FOUND is not
+dnl	specified
```
Is there some weird indentation here?

ben-palmer-sociomantic commented on docs/latex_check_packages.sh:
```diff
@@ -0,0 +1,97 @@
+  printf "%-22s " $f;
+  FOUND="Yes"
+  # adds package to MISSING packages if the can't be found
+  # by  neither kpsewhich nor find
```
extra space between `by` and `neither`

ben-palmer-sociomantic reviewed (commented):
Couple of minor comments on documentation, otherwise LGTM.

andreas-bok-sociomantic commented:
Fixed weird indentation. 
The build fails on travis due to an error caused by the mysql service being started.

andreas-bok-sociomantic commented:
The build error could be related to this change at Travis https://docs.travis-ci.com/user/build-environment-updates/2016-12-01/#Details, with reported issues https://github.com/travis-ci/travis-ci/issues/6961

andreas-bok-sociomantic committed:
Fix filename of generated pdf documentation

andreas-bok-sociomantic commented:
Updated
- fixed a few errors, 
- rebased on newest v5.x.x branch

ben-palmer-sociomantic commented:
LGTM
